storage/filesystem/internal/dotgit/dotgit.go

```diff
@@ -502,11 +503,29 @@ func (d *DotGit) readReferenceFile(path, name string) (ref *plumbing.Reference,
 	return plumbing.NewReferenceFromStrings(name, line), nil
 }
 
-// Module return a billy.Filesystem poiting to the module folder
+// Module return a billy.Filesystem pointing to the module folder
 func (d *DotGit) Module(name string) (billy.Filesystem, error) {
 	return d.fs.Chroot(d.fs.Join(modulePath, name))
 }
 
+// Alternates returns the content of objects/info/alternates if available.
+// This can be used to checks if it's a shared repository.
+func (d *DotGit) Alternates() (string, error) {
```

Comment:
Maybe it would be better if this returned `(*DotGit, error)` instead of just the file contents?